CMakeLists.txt

```diff
@@ -321,6 +321,8 @@ add_subdirectory(src)
 
 if (BENCHMARK_ENABLE_TESTING)
   enable_testing()
+  # The test directory must be added before include(GoogleTest). GoogleTest.cmake sets -w globally.
+  add_subdirectory(test)
```

Comment:
Does CMake support delayed target parsing?
In the sense, tests in `test/` link to `gtest`/`gtest_main`/`gmock`/`gmock_main` targets,
but said targets won't exist yet, because we now add tests before creating them.
Does that work in CMake? I've literally never seen that before.

Reply:
It does work for `target_link_libraries`. It does not work for modifying the target. E.g., adding `target_compile_options (gtest PRIVATE "-w")` in `test/CMakeLists.txt` will fail because the `gtest` target is not yet defined.

Alternatively, the `-w` could only be set for the gtest/gmock targets.

Reply:
I think we should fix the `-w` bug while we are at it, globally, not just for our particular problem.
Could you please try changing the current `add_compile_options(-w)` logic to the `target_compile_options()`?

Reply:
> Could you please try changing the current add_compile_options(-w) logic to the target_compile_options()?

(i'm **hoping** it would just work, and not uncover hundreds of unintentionally-silenced warnings...)

Reply:
With the current change, the warnings are already unsilenced. I didn't get any warnings for the whole test suite locally. So far so good :)

I'll only be able to incorporate the changes this afternoon (which is in 6 hours for me).

Reply:
Done